src/Controller/CRUDController.php

```diff
@@ -1521,12 +1429,10 @@ private function setFormTheme(FormView $formView, array $theme = null): void
         $twig->getRuntime(FormRenderer::class)->setTheme($formView, $theme);
     }
 
-    private function handleXmlHttpRequestErrorResponse(Request $request, FormInterface $form): ?JsonResponse
+    private function handleXmlHttpRequestErrorResponse(Request $request, FormInterface $form): JsonResponse
     {
         if ('application/json' !== $request->headers->get('Accept')) {
-            @trigger_error('In next major version response will return 406 NOT ACCEPTABLE without `Accept: application/json`', E_USER_DEPRECATED);
-
-            return null;
+            return $this->renderJson([], Response::HTTP_NOT_ACCEPTABLE);
```

Comment:
Maybe this change should be made in the stable branch first, but the `Accept` header could allow "application/json" and at the same time other types:
>application/json, application/xml;q=0.9, \*/\*;q=0.8

Even, it could include extra data, like API versioning:
>application/json;version=2.0

Reply:
I've created #5962 to solve this.